File: solutions/typescript/0035.ts
function searchInsert(nums: number[], target: number): number {
    let left = 0;
    let right = nums.length - 1;
    
    while (left <= right) {
        const mid = Math.floor((left + right) / 2);
        
        if (nums[mid] === target) {
            return mid; // 找到目标值，返回索引
        } else if (nums[mid] < target) {
            left = mid + 1; // 目标值在右半部分
        } else {
            right = mid - 1; // 目标值在左半部分
        }
    }
    
    // 如果没找到目标值，left 就是应该插入的位置
    return left;
};

File: solutions/rust/0035.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        if nums.is_empty() {
            return 0;
        }
        let mut left: isize = 0;
        let mut right: isize = nums.len() as isize - 1;
        while left <= right {
            let mid = left + (right - left) / 2;
            let val = nums[mid as usize];
            if val == target {
                return mid as i32;
            } else if val < target {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        left as i32
    }
}